Pod/Core/Step.swift

```diff
@@ -8,24 +8,41 @@
 
 import Foundation
 
-protocol StepFunctionParameters {}
+public struct StepMatches<T: MatchedStringRepresentable> {
+    public let allMatches: [T]
```

Comment:
Should these be public if we are controlling access via the subscript methods (imho they shouldn't be, I'd make them private :)

Reply:
Yes, we can expose one or another but IMO there is no point in private subscripts, so I'll make properties private and leave subscripts public, they are also nices from user point of view

Reply:
Sorry, I didn't explain myself clearly - that's exactly what I thought, private properties and public subscripts :)